x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/action/filter/ShardBulkInferenceActionFilter.java

```diff
@@ -489,6 +519,10 @@ private long addFieldInferenceRequests(BulkItemRequest item, int itemIndex, Map<
                          * This ensures that the field is treated as intentionally cleared,
                          * preventing any unintended carryover of prior inference results.
                          */
+                        if (incrementIndexingPressure(indexRequest, itemIndex) == false) {
+                            return inputLength;
```

Comment:
Once this happened should all the subsequent bulk items be failed?  There is a chance that indexing pressure may have dropped sufficiently to allow a later request but does complicate the work the caller has to do processing rejected items in the bulk response?